src/main/kotlin/com/fasterxml/jackson/module/kotlin/Extensions.kt

```diff
@@ -121,8 +121,20 @@ internal fun Int.toBitSet(): BitSet {
     return bits
 }
 
+private val jvmInlineClass by lazy {
+    try {
+        Class.forName("kotlin.jvm.JvmInline")
+    } catch (e: ClassNotFoundException) {
+        null
+    }
+}
+
 // In the future, value classes without @JvmInline will be available, and unboxing may not be able to handle it.
 // https://github.com/FasterXML/jackson-module-kotlin/issues/464
 // The JvmInline annotation can be added to Java classes,
 // so the isKotlinClass decision is necessary (the order is preferable in terms of possible frequency).
-internal fun Class<*>.isUnboxableValueClass() = annotations.any { it is JvmInline } && this.isKotlinClass()
+internal fun Class<*>.isUnboxableValueClass() = if (jvmInlineClass != null) {
+    annotations.any { jvmInlineClass!!.isInstance(it) } && this.isKotlinClass()
```

Comment:
Huh, I'm surprised the Kotlin compiler can't null-track that properly

Reply:
Yea, it's because it's a field on an interface, not a data class. So it's nullability could, theoretically, change between callsite and usage.